src/charts/chart.js

```diff
@@ -249,19 +246,12 @@ export default function Chart (_container) {
         brushRangeEditor: BrushRangeEditor(cache.headerGroup),
         label: Label(cache.root),
         clipPath: ClipPath(cache.svg)
-      }
-
-      eventCollector = {
-        onBrush: rebind(components.brush),
-        onHover: rebind(components.hover),
-        onBinning: rebind(components.binning),
-        onDomainEditor: rebind(components.domainEditor),
-        onBrushRangeEditor: rebind(components.brushRangeEditor),
-        onLabel: rebind(components.label),
-        onPanel: rebind(dispatcher)
-      }
+      })
     }
+    return this
+  }
 
+  function update () {
```

Comment:
I splitted `build` from `update`

Reply:
makes sense

Reply:
is it intended that this will be the case for each component now as well?

Reply:
I don't think so. The main component is special in that it needs to init before everything else and update when it's time to render all components